lib/bolt_server/transport_app.rb

```diff
@@ -188,6 +188,11 @@ def run_script(target, body)
       [200, GC.stat.to_json]
     end
 
+    get '/admin/status' do
+      stats = Puma.stats
```

Comment:
This seems much simpler! I'm not sure I follow what you mean by the threading part... Also tests are not liking this constant. Might need a require or to fully qualify this.

Reply:
The default status app is started here:

https://github.com/puma/puma/blob/master/lib/puma/runner.rb#L49

Its "out of band" as in it has its own thread pool (from ace or bolt) ... limited to 1 thread.

Reply:
I noticed that we don't test the other (`/admin/gc_stat`) stat endpoint:

https://github.com/puppetlabs/bolt/blob/master/lib/bolt_server/transport_app.rb#L187

And `stats` is tested in puma:

https://github.com/puma/puma/blob/master/test/test_app_status.rb

Rather than a `require` I think I would have to mock `Puma.stats` resulting in a test that just checks this:

```
stats.is_a?(Hash) ? stats.to_json : stats
```

Possibly we do not have to test `/admin/status`, since we don't test `/admin/gc_stat` ? 

Here is a real test in Vagrant:

```
[root@pe-201921-master-ha ~]# curl https://$(puppet config print server):62658/admin/status --cacert /etc/puppetlabs/puppet/ssl/certs/ca.pem --cert /etc/puppetlabs/puppet/ssl/certs/$(puppet config print certname).pem --key /etc/puppetlabs/puppet/ssl/private_keys/$(puppet config print certname).pem
{ "backlog": 0, "running": 1, "pool_capacity": 99, "max_threads": 100 }
```